torch/_inductor/utils.py

```diff
@@ -1215,8 +1215,8 @@ def is_linux() -> bool:
     return platform.system() == "Linux"
 
 
-def has_free_symbols(itr):
-    return any(hasattr(x, "free_symbols") and len(x.free_symbols) > 0 for x in itr)
+def has_free_symbols(itr: Iterable[Any]):
+    return any(isinstance(x, sympy.Expr) and not x.is_number for x in itr)
```

Comment:
A non-equivalent definition? Perhaps better to change the name of this one or try to remove it?

Reply:
It's not equivalent in general, but I think that they may be equivalent in their callsites.

Reply:
I think it's mainly used by the `is_dynamic` function defined below.  In which case it's close enough in space to eliminate confusion.